cardscontainer.cpp

```diff
@@ -0,0 +1,61 @@
+#include "cardscontainer.h"
+
+#include <QVariant>
+#include <iostream>
+
+CardsContainer::CardsContainer(QWidget* parent) :
+    QWidget(parent),
+    parent_(parent)
+{
+
+    createLayout(parent_);
+
+    displayCards();
+}
+
+void CardsContainer::displayCards(){
+
+        int i = 0;
+        for(auto& it : cards_){
+            showCard("../../BakcylProgramowania_2020/source/pic.png",it); //TODO: Zmienić tego statycznego stringa na listę zdjęć i iterować po zdjęciach a nie po buttonach
+            it->setProperty("Id", i);                                     //TODO: ID powinno być id karty
+            i++;
+        }
+}
+
+void CardsContainer::createLayout(QWidget* parent)
+{
+
+    int posX = 90;
+    int posY = 30;
+
+    const size_t cardHeight = 160;
+    const size_t cardWight = 120;
+
+    const size_t maxCardsDisplay = 15;
+
+    for(size_t i = 0; i < maxCardsDisplay; i++){
+
+        cards_.push_back(new QPushButton(parent));
+        cards_[i]->move(posX,posY);
```

Comment:
czemu nie .at(i) i jaką mamy pewność, że pod indeksem [i] znajduje się legitny obiekt? Chodzi mi o to, żeby się nie okazało że coś sie scrashuje, bo pod cards_[i} nic nie będzie

Reply:
.at(i) spowalnia dostęp i nie możliwe by nic tam nie było chyba, że samo tworzenie QPushButton się nie powiedzie